### PR TITLE
Heading bug fix mcf2

### PR DESCRIPTION
**Pull request by mcfogarty**
Apologies for undoing commits; should have tested in another branch.

Heading, pitch, roll outputs were checked for ADV w/IMU bench tests, as well as field data for ADV w/IMU on SMB and RDI workhorse on SMB. All calculations make sense.

Edits include fixing a few typos and adding (or requesting) more detail in comments.

### CODE REVIEW

File: dolfyn/io/rdi.py [hunk 1]
```diff
@@ -41,15 +41,15 @@ def read_rdi(fname, userdata=None, nens=None, keep_orient_raw=False):
     adp_data : :class:`ADPdata <dolfyn.adp.base.ADPdata>`
 
     """
-    # userdata is not used here.
+    # userdata is not used here. CLARIFY COMMENT..
```
Comment (lkilcher) [inline dolfyn/io/rdi.py]: That's an old comment. Delete this line.

File: dolfyn/io/rdi.py [hunk 2]
```diff
@@ -41,15 +41,15 @@ def read_rdi(fname, userdata=None, nens=None, keep_orient_raw=False):
...
     userdata = read_userdata(fname, userdata)
     with adcp_loader(fname) as ldr:
         dat = ldr.load_data(nens=nens)
     dat['props'].update(userdata)
 
     od = dat['orient']
 
-    # value in userdata, otherwise None
+    # value in userdata, otherwise None. CLARIFY COMMENT..
```
Comment (lkilcher) [inline dolfyn/io/rdi.py]: Old comment. Delete this line.

File: dolfyn/rotate/base.py [hunk 3]
```diff
@@ -77,11 +86,12 @@ def euler2orient(heading, pitch, roll, units='degrees'):
        - a "ZYX" rotation order. That is, these variables are computed
          assuming that rotation from the earth -> instrument frame happens
          by rotating around the z-axis first (heading), then rotating
-         around the y-axis (pitch), then rotating around the x-axis (roll).
+         around the y-axis (pitch), then rotating around the x-axis (roll). 
+         Note this requires matrix multiplication in the reverse order.
```
Comment (lkilcher) [inline dolfyn/rotate/base.py]: Good note! Why don't you also add this as a comment above `return np.einsum(...)` at the end of this function. Something like:

    # As mentioned in the docs, the matrix-multiplication order is "reversed" (i.e., ZYX
    # order of rotations happens by multiplying R*P*H).
    # It helps to think of this as left-multiplying omat onto a vector. In which case,
    # H gets multiplied first, then P, then R (i.e., the ZYX rotation order).

File: dolfyn/rotate/base.py [hunk 4]
```diff
@@ -138,13 +151,18 @@ def orient2euler(omat):
...
+      positive clockwise from North (this is *opposite* the right-hand-rule
+      around the Z-axis), range 0-360 degrees.
     pitch : np.ndarray
-      The pitch angle of the ADV (degrees).
+      The pitch angle (degrees). Pitch is positive when the x-axis 
+      pitches up (this is *opposite* the right-hand-rule around the Y-axis).
     roll : np.ndarray
-      The pitch angle of the ADV (degrees).
+      The roll angle (degrees). Roll is positive according to the 
+      right-hand-rule around the instument's x-axis.
```
Comment (lkilcher) [inline dolfyn/rotate/base.py]: These are excellent!

File: dolfyn/rotate/rdi.py [hunk 5]
```diff
@@ -148,8 +149,14 @@ def inst2earth(adcpo, reverse=False,
...
 def calc_orientmat(adcpo):
 
-    # Now calculate the rotation matrix.
+    # Calculate the orientation matrix using the raw 
+    # heading, pitch, roll values from the RDI binary file.
+
     """
+     Parameters
+    ----------
+    -adcpo : The ADP object containing the data.
```
Comment (lkilcher) [inline dolfyn/rotate/rdi.py]: remove the `-` at the beginning of this line.

File: dolfyn/rotate/vector.py [hunk 6]
```diff
@@ -148,6 +148,8 @@ def calc_omat(hh, pp, rr, orientation_down=None):
         #       orientation corresponds to the communication cable
         #       being up.  This is ridiculous, but apparently a
         #       reality.
+        #       In DOLfYN 'orientation_down' = True indicates the ADV probe head
+        #       is pointing UP.
```
Comment (lkilcher) [inline dolfyn/rotate/vector.py]: I think the correct text here is: "Nortek defines 'down' orientation as the ADV probe head points up."

File: dolfyn/io/rdi.py (hunk 1 above)
Comment (mcfogarty) [reply dolfyn/io/rdi.py]: Done.

File: dolfyn/io/rdi.py (hunk 2 above)
Comment (mcfogarty) [reply dolfyn/io/rdi.py]: Done.

File: dolfyn/rotate/base.py (hunk 3 above)
Comment (mcfogarty) [reply dolfyn/rotate/base.py]: Done.

File: dolfyn/rotate/rdi.py (hunk 5 above)
Comment (mcfogarty) [reply dolfyn/rotate/rdi.py]: Done.

File: dolfyn/rotate/vector.py (hunk 6 above)
Comment (mcfogarty) [reply dolfyn/rotate/vector.py]: I didn't change anything here. It was confusing to me to try to sort out what 'orientation_down' in DOLfYN meant; I wasn't sure if it was still in Nortek defined terms or in down-means-down terms. I think the documentation needs a section clearly explaining some key details about ADVs without IMUs that are important to understanding DOLfYN plus a few links to more info in general ---  'orientation_down' is an output field from the binary file only from an ADV without an IMU, it'll = True if the ADV probe was headed up (Nortek Status bit 0 = 1) and = False (Nortek Status bit 0 = 0) if it the head was pointed down, that it only gets defined on the initial measurement in a time series, etc.. Reference the Transform.m file from Nortek as well as https://www.nortekgroup.com/faq/what-do-the-error-and-status-codes-mean, and Sven's 9/1/2009 posting on  http://www.nortek.no/en/knowledge-center/forum/current-profilers-and-current-meters/579860304 "You can read more about this in the manual, but what you need to be aware of is the update of the up/down orientation of the system. In order to calculate correct ENU coordinates there is an up/down sensor in the Vector as a part of the tilt sensor. And if you run continuous measurements this orientation is only established at the time when the deployment starts. "

**Comment by mcfogarty** [comment]
FYI when 
dat = dolfyn.read(fname_rdi + '.000', keep_orient_raw = True), 
the heading, pitch, and roll end up in dat.orient.raw, but heading_std, pitch_std, roll_std remain in dat.orient

**Comment by lkilcher** [comment]
Merging into my PR #50

**Pull request merged**